compiler/rustc_middle/src/ty/mod.rs

```diff
@@ -131,6 +131,8 @@ pub struct ResolverOutputs {
     pub definitions: rustc_hir::definitions::Definitions,
     pub cstore: Box<CrateStoreDyn>,
     pub visibilities: FxHashMap<LocalDefId, Visibility>,
+    /// This field is used to decide whether we should make `PRIVATE_IN_PUBLIC` a hard error.
+    pub has_pub_restricted: bool,
```

Comment:
Shouldn't it be included into the crate hash too?

Reply:
It is only used to emit a diagnostic, which won't appear in metadata.  Therefore, it does not need to appear in `crate_hash`.  I can include it as a safety precaution.